eng/mgmt/automation/utils.py

```diff
@@ -0,0 +1,179 @@
+    all_module.add(module)
+    all_module = [
+        indent + POM_MODULE_FORMAT.format(module)
+        for module in sorted(all_module)
+    ]
+
+    return '<modules>\n' + ''.join(all_module) + post_module.group()
+
+
+def add_module_to_default_profile(pom: str, module: str) -> Tuple[bool, str]:
+    for profile in re.finditer(r'<profile>[\s\S]*?</profile>', pom):
+        profile_value = profile.group()
+        if re.search(r'<id>default</id>', profile_value):
+            if len(re.findall('<modules>', profile_value)) > 1:
+                logging.error(
+                    '[POM][Profile][Skip] find more than one <modules> in <profile> default'
+                )
+                return (False, '')
+            modules = re.search(r'<modules>[\s\S]*</modules>', profile_value)
+            if not modules:
+                logging.error(
+                    '[POM][Profile][Skip] Cannot find <modules> in <profile> default'
+                )
+                return (False, '')
+            modules_update = add_module_to_modules(modules.group(), module)
+            pre_modules = pom[:profile.start() + modules.start()]
+            post_modules = pom[profile.start() + modules.end():]
+            return (True, pre_modules + modules_update + post_modules)
+    logging.error(
+        '[POM][Profile][Skip] cannot find <profile> with <id> default')
+    return (False, '')
+
+
+def add_module_to_pom(pom: str, module: str) -> Tuple[bool, str]:
+    if pom.find('<module>{0}</module>'.format(module)) >= 0:
+        logging.info('[POM][Skip] pom already has module {0}'.format(module))
+        return (True, pom)
+
+    if len(re.findall('<modules>', pom)) > 1:
+        if pom.find('<profiles>') >= 0:
+            return add_module_to_default_profile(pom, module)
+        logging.error('[POM][Skip] find more than one <modules> in pom')
+        return (False, '')
+
+    modules = re.search(r'<modules>[\s\S]*</modules>', pom)
+    if not modules:
+        logging.error('[POM][Skip] Cannot find <modules> in pom')
+        return (False, '')
+
+    modules_update = add_module_to_modules(modules.group(), module)
+    pre_modules = pom[:modules.start()]
+    post_modules = pom[modules.end():]
+    return (True, pre_modules + modules_update + post_modules)
+
+
+def update_root_pom(sdk_root: str, service: str):
+    pom_file = os.path.join(sdk_root, 'pom.xml')
+    if not os.path.exists(pom_file):
+        logging.error('[POM][Skip] cannot find root pom')
+        return
+
+    module = 'sdk/{0}'.format(service)
+    with open(pom_file, 'r') as fin:
+        pom = fin.read()
+
+    logging.info('[POM][Process] dealing with root pom')
+    success, pom = add_module_to_pom(pom, module)
+    if success:
+        with open(pom_file, 'w') as fout:
+            fout.write(pom)
+        logging.info('[POM][Success] Write to root pom')
+
+
+def update_service_ci_and_pom(sdk_root: str, group: str, service: str, module: str):
```

Comment:
mostly moved from `generate.py`, with minor refactor to allow it handle data-plane SDK as well.